docs/connect/clients/java.md

```diff
@@ -187,104 +187,126 @@ public class Main {
-
-class User {
-    private String name;
-    private String email;
-    private int age;
-    private String city;
-
-    public User(String name, String email, int age, String city) {
-        this.name = name;
-        this.email = email;
-        this.age = age;
-        this.city = city;
-    }
-
-    //...
-}
-```
+1. A connection is requested from the pool.
+2. A connection is served. 
+   - An idle connection is served when there are available non-active connections or, 
+   - A connection is created when the number of connections is under maxTotal. 
+3. The connection becomes active.
+4. The connection is released back to the pool.
+5. The connection is marked as stale. 
+6. The connection is kept idle for `minEvictableIdleTime`. 
+7. The connection becomes evictable if the number of connections is greater than `minIdle`. 
+8. The connection is ready to be closed.
 
-Connect to your Redis database with `JedisPooled`.
+It's important to configure connection pool correctly. 
+To configure the connection pool, use `GenericObjectPoolConfig` from [Apache Commons Pool2](https://commons.apache.org/proper/commons-pool/apidocs/org/apache/commons/pool2/impl/GenericObjectPoolConfig.html).
 
-```java
-JedisPooled jedis = new JedisPooled("localhost", 6379);
-```
 
-Let's create some test data to add to your database.
 
 ```java
-User user1 = new User("Paul John", "paul.john@example.com", 42, "London");
-User user2 = new User("Eden Zamir", "eden.zamir@example.com", 29, "Tel Aviv");
-User user3 = new User("Paul Zamir", "paul.zamir@example.com", 35, "Tel Aviv");
+
+ConnectionPoolConfig poolConfig = new ConnectionPoolConfig();
+// maximum active connections in the pool,
+// tune this according to your needs and application type
+// default is 8
+poolConfig.setMaxTotal(8);
+
+// maximum idle connections in the pool, default is 8
+poolConfig.setMaxIdle(8);
+// minimum idle connections in the pool, default 0
+poolConfig.setMinIdle(0);
+
+// Enables waiting for a connection to become available.
+poolConfig.setBlockWhenExhausted(true);
+// The maximum number of seconds to wait for a connection to become available
+poolConfig.setMaxWait(Duration.ofSeconds(1));
+
+// Enables sending a PING command periodically while the connection is idle.
+poolConfig.setTestWhileIdle(true);
+// controls the period between checks for idle connections in the pool
+poolConfig.setTimeBetweenEvictionRuns(Duration.ofSeconds(1));
+
+// JedisPooled does all hard work on fetching and releasing connection to the pool
+// to prevent connection starvation
+JedisPooled jedis = new JedisPooled(poolConfig, "localhost", 6379);
 ```
 
-Create an index. In this example, all JSON documents with the key prefix `user:` are indexed. For more information, see [Query syntax](/docs/interact/search-and-query/query/).
 
+### Timeout
+
+To set a timeout for a connection, use `JedisPooled` or `JedisPool` constructor with `timeout` parameter or `JedisClientConfig` with `socketTimeout` parameter:

```

Comment:
There are several types of timeout.
- connectionTimeout
- soTimeout / socketTimeout
- infiniteSoTimeout / blockingSocketTimeout (socketTimeout but for blocking commands)

Lone timeout parameter in constructors acts value for both connectionTimeout and socketTimeout.

Reply:
Fixed